staging/src/kubevirt.io/client-go/api/v1/types.go

```diff
@@ -1334,14 +1351,21 @@ type SMBiosConfiguration struct {
 // MigrationConfiguration holds migration options
 // +k8s:openapi-gen=true
 type MigrationConfiguration struct {
-	AllowAutoConverge                 bool               `json:"allowAutoConverge,string"`
-	BandwidthPerMigration             *resource.Quantity `json:"bandwidthPerMigration,omitempty"`
-	CompletionTimeoutPerGiB           *int64             `json:"completionTimeoutPerGiB,string,omitempty"`
-	NodeDrainTaintKey                 *string            `json:"nodeDrainTaintKey,omitempty"`
-	ParallelOutboundMigrationsPerNode *uint32            `json:"parallelOutboundMigrationsPerNode,string,omitempty"`
-	ParallelMigrationsPerCluster      *uint32            `json:"parallelMigrationsPerCluster,string,omitempty"`
-	ProgressTimeout                   *int64             `json:"progressTimeout,string,omitempty"`
-	UnsafeMigrationOverride           bool               `json:"unsafeMigrationOverride,string"`
+	NodeDrainTaintKey                 *string `json:"nodeDrainTaintKey,omitempty"`
+	ParallelOutboundMigrationsPerNode *uint32 `json:"parallelOutboundMigrationsPerNode,string,omitempty"`
+	ParallelMigrationsPerCluster      *uint32 `json:"parallelMigrationsPerCluster,string,omitempty"`
+	*PerVMIMigrationConfiguration     `json:",inline"`
+}
+
+// PerVMIMigrationConfiguration holds migration options that can be configurated per vmi
+// +k8s:openapi-gen=true
+type PerVMIMigrationConfiguration struct {
+	AllowAutoConverge       *bool              `json:"allowAutoConverge,string,omitempty"`
+	BandwidthPerMigration   *resource.Quantity `json:"bandwidthPerMigration,omitempty"`
+	CompletionTimeoutPerGiB *int64             `json:"completionTimeoutPerGiB,string,omitempty"`
+	ProgressTimeout         *int64             `json:"progressTimeout,string,omitempty"`
+	UnsafeMigrationOverride *bool              `json:"unsafeMigrationOverride,string,omitempty"`
+	UsePostCopy             *bool              `json:"usePostCopy,string,omitempty"`
```

Comment:
should the new ```MigrationMode``` be used to indicate this now?

Reply:
do you mean instead of a boolean specify which migration to use? I thought it might make more sense to keep as a boolean since the flag to use postcopy in a migration is a boolean, but we could have the user explicitly set `PostCopy` or `PreCopy`. Should it then be required?

Reply:
it could be left optional and we default to precopy when no value is set.

I only brought this up because of the "mode" section on the VMI status. It seems like we'd want to use consistent ways across the api for representing pre/post copy